### PR TITLE
Add capybara and headless poltergeist gems for feature specs and write test

### PR DESCRIPTION
**Pull request by Johnnycus**
Capybara and selenium gems for testing. Tests that enters text in React's inputs and ensures that text is displayed.

### CODE REVIEW

**Comment by justin808** [comment]
@Johnnycus Please use as examples the tests in https://github.com/shakacode/react-webpack-rails-tutorial/tree/master/spec/features and https://github.com/reactjs/react-rails/tree/master/test.

However, we're using rspec and not minitest.

**Comment by justin808** [comment]
@Johnnycus Pay careful attention to: https://github.com/jnicklas/capybara#using-capybara-with-rspec for the correct API for integration tests. @samnang if you can give Albert a hand, he might write a bunch of tests for us. There's some good tips here: http://betterspecs.org/.

**Comment by samnang** [comment]
@Johnnycus Should we try lightweight headless browser like phanthomjs with https://github.com/jnicklas/capybara#poltergeist?

We could have different test levels here. Unit level to test isolation just for ruby side, and integration tests that should live under dummy app. We suppose to have integration tests with different scenarios, we shouldn't it renders on the example page of dummy app. I'm not sure yet, but I suppose we have different   stub files that we could pull and use it in this, then we will call the rails helper to render the component inside that file, then we expect the output.

**Comment by Johnnycus** [comment]
@samnang Yes, thanks for an advice, I'm going to add poltergeist.

**Comment by Johnnycus** [comment]
@justin808 Ok, thanks!

**Comment by Johnnycus** [comment]
@samnang Now we have headless browser for testing.

File: spec/dummy/vendor/assets/javascripts/es5-shim.js [hunk 1]
```diff
@@ -0,0 +1,1596 @@
+/*!
```
Comment (justin808) [inline spec/dummy/vendor/assets/javascripts/es5-shim.js:1]: @Johnnycus We add dependencies via NPM. Please remove this file and add via npm. See https://github.com/shakacode/react-webpack-rails-tutorial/blob/master/client/package.json

**Comment by justin808** [comment]
Closing for now. @Johnnycus Feel free to pull the latest and submit some more tests. Tests are **always helpful**. Let us know if we can help you. Thanks for helping!

**Pull request closed without merging**